firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java

```diff
@@ -132,4 +140,55 @@ private PublicKey parseKey(String base64PublicKey) {
     }
     return null;
   }
+
+  private static String getDefaultSenderId(FirebaseApp app) {
+    // Check for an explicit sender id
+    String senderId = app.getOptions().getGcmSenderId();
+    if (senderId != null) {
+      return senderId;
+    }
+    String appId = app.getOptions().getApplicationId();
+    if (!appId.startsWith("1:")) {
+      // Not v1, server should be updated to accept the full app ID now
+      return appId;
```

Comment:
I don't understand this clause.
Why should the App-ID start with something different than ":1"?
What does this comment mean? "server should be updated to accept the full app ID now"